Sources/Hooks/Hook/UseEffect.swift

```diff
@@ -63,18 +63,18 @@ private struct EffectHook: Hook {
     }
 
     func updateState(coordinator: Coordinator) {
+        coordinator.state.cleanup?()
         coordinator.state.cleanup = effect()
     }
 
     func dispose(state: State) {
+        state.cleanup?()
         state.cleanup = nil
     }
 }
 
 private extension EffectHook {
     final class State {
-        var cleanup: (() -> Void)? {
-            didSet { oldValue?() }
-        }
```

Comment:
Note: Updating code to `willSet` is not enough, because `effect()` was already executed.